imports/src/parser/import_parser.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2019-2020 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::errors::ImportParserError;
+use leo_typed::{Package, Program};
+
+use std::{collections::HashMap, env::current_dir};
+
+/// Stores imported packages.
+///
+/// A program can import one or more packages. A package can be found locally in the source
+/// directory, foreign in the imports directory, or part of the core package list.
+#[derive(Clone)]
+pub struct ImportParser {
+    imports: HashMap<String, Program>,
+    core_packages: Vec<Package>,
+}
+
+impl ImportParser {
+    ///
+    /// Creates a new empty `ImportParser`.
+    ///
+    pub fn new() -> Self {
+        Self {
+            imports: HashMap::new(),
+            core_packages: vec![],
+        }
+    }
+
+    ///
+    /// Inserts a (file name -> program) pair into the `ImportParser`.
+    ///
+    /// It is okay if the imported program is already present since importing multiple symbols from
+    /// the same file is allowed.
+    ///
+    pub(crate) fn insert_import(&mut self, file_name: String, program: Program) {
+        // Insert the imported program.
+        let _program = self.imports.insert(file_name.clone(), program);
+    }
+
+    ///
+    /// Inserts a core package into the `ImportParser`.
+    ///
+    /// If the vector did not have this file_name present, `Ok()` is returned.
+    ///
+    /// If the vector did have this file_name present, a duplicate import error is thrown.
+    ///
+    pub(crate) fn insert_core_package(&mut self, package: &Package) -> Result<(), ImportParserError> {
+        // Check for duplicate core package name.
+        if self.core_packages.contains(package) {
```

Comment:
if it is expected that the number of packages is low, it's ok to be using a `Vec` for `core_packages`, but if it is expected to be higher, e.g. a `HashSet` should be used instead